src/ds/d500/d500-auto-calibration.cpp

```diff
@@ -224,19 +182,11 @@ namespace librealsense
     std::vector<uint8_t> d500_auto_calibrated::update_abort_status()
     {
         std::vector<uint8_t> ans;
-        auto res = _hw_monitor->send(command{ ds::GET_CALIB_STATUS });
-        if (!check_buffer_size_from_get_calib_status(res))
-            throw std::runtime_error("GET_CALIB_STATUS returned struct with wrong size");
-
-        d500_calibration_answer calib_answer = *reinterpret_cast<d500_calibration_answer*>(res.data());
+        auto calib_answer = _ac_handler->get_status();
         if (calib_answer.calibration_state == static_cast<uint8_t>(d500_calibration_state::RS2_D500_CALIBRATION_STATE_PROCESS))
         {
             std::this_thread::sleep_for(std::chrono::milliseconds(200));
-            res = _hw_monitor->send(command{ ds::GET_CALIB_STATUS });
-            if (!check_buffer_size_from_get_calib_status(res))
-                throw std::runtime_error("GET_CALIB_STATUS returned struct with wrong size");
-
-            calib_answer = *reinterpret_cast<d500_calibration_answer*>(res.data());
+            calib_answer = _ac_handler->get_status();
```

Comment:
Notice that D555e may have auto calibration process in the future.
I still don't know what it means, but if it will need classes _ac_ might be confusing.
You can double check with Eran

Reply:
Will check

Reply:
thought about that - I think even if we decide to use some other auto calibration process later on, this refactoring can still be relevant for it - so this _ac_handler will still remain the same.
It only provides a way to implement the calibration actions via USB (hwm) or via DDS